api/src/main/resources/api/service_openapi.yaml

```diff
@@ -20,8 +20,149 @@ paths:
         500:
           description: Service is broken
 
+  # TODO implement "real" export endpoints
+  '/entityFilter/sqlQuery':
+    post:
+      # TODO authorization
+      security: []
+      summary: Returns an SQL query for listing all of a Entity's primary keys.
+      operationId: entitySqlQuery
+      tags: [Entity]
+      requestBody:
+        required: true
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/EntityFilter'
+      responses:
+        200:
+          $ref: '#/components/responses/SqlQueryResponse'
+
 components:
+  responses:
+    SqlQueryResponse:
+      description: A raw SQL query.
+      content:
+        application/json:
+          schema:
+            $ref: '#/components/schemas/SqlQuery'
+
   schemas:
+    AttributeVariable:
+      description: An attribute of a variable for an entity.
+      type: object
+      required: [name, variable]
+      properties:
+        name:
+          description: The name of the attribute.
+          type: string
+        variable:
+          description: The name of the variable.
+          type: string
+
+    AttributeValue:
+      type: object
+      description: A possible value of an attribute. Exactly one field must be specified.
+      properties:
+        int64Val:
+          type: integer
+          format: int64
+        stringVal:
+          type: string
+        boolVal:
+          type: boolean
+
+    EntityFilter:
+      description: A filter to apply to the given entity.
+      type: object
+      properties:
+        entity:
+          description: The name of the entity
+          type: string
+        underlay:
+          description: The name of the underlay containing this entity.
+          type: string
+        entityVariable:
+          description: The variable to use for the top level entity in the filter.
+          type: string
+        filter:
+          $ref: '#/components/schemas/Filter'
+
+    SqlQuery:
```

Comment:
Possibly not in this version yet, but should this give some indication of the specific database instance, or at least database type that it has been generated for?

Also, probably not for this version, do we need parameters here?

Reply:
Yep, added a few TODOs.
I'm not sure we keep this as a meaningful kind of "export"
I'm not sure that parameters help any, because if you don't trust the query you get from us, why should you trust that we have parameterized it correctly.